projects/batfish/test/org/batfish/main/BatfishTest.java

```diff
@@ -0,0 +1,68 @@
+package org.batfish.main;
+
+import org.batfish.common.BatfishException;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+
+public class BatfishTest {
+
+    @Test(expected = BatfishException.class)
+    public void testNotExistingPath() {
+        List<Path> result = Batfish.listAllFiles(Paths.get("./test/notExist"));
+    }
+
+
+    @Test
+    public void testNoFileUnderPath() {
+        List<Path> result = Batfish.listAllFiles(Paths.get("./test/noFile"));
+        Assert.assertEquals(0, result.size());
+    }
+
+
+    @Test
+    public void testReadUnNestedPath() {
+        List<Path> result = Batfish.listAllFiles(Paths.get("./test/unNestedDirectory"));
+        Assert.assertEquals(3, result.size());
+    }
+
+
+    @Test
+    public void testReadStartWithDotFile() {
+        List<Path> result = Batfish.listAllFiles(Paths.get("./test/startWithDot"));
+        Assert.assertEquals(0, result.size());
+    }
+
+
+    @Test
+    public void testFileNameRead() {
+        List<Path> result = Batfish.listAllFiles(Paths.get("./test/unNestedDirectory"));
+        Assert.assertEquals("./test/unNestedDirectory/test1.cfg", result.get(0).toString());
```

Comment:
here it should be something like:

```java
List<Path> expected = // build a list here of the expected paths.
assertThat(result, equalTo(expected));
```

(We use `equalTo` instead of `containsInAnyOrder` since the function is supposed to return sorted output and thus always the same answer.)